projects/tinkerbell/hook/patches/0001-Make-registry-certificate-hosting-URL-and-tink-worke.patch

```diff
@@ -0,0 +1,125 @@
+  defaulting to the hardcoded default of 'tink-worker:latest', use the
+  TINK_WORKER_IMAGE environment variable setting in boots. Boots passes this over
+  to hook via command-line parameter 'tink_worker_image'
+---
+ bootkit/main.go     | 19 ++++++++++++++++++-
+ tink-docker/main.go | 12 +++++++++++-
+ 2 files changed, 29 insertions(+), 2 deletions(-)
+
+diff --git a/bootkit/main.go b/bootkit/main.go
+index 48ba577..5f5f3bc 100644
+--- a/bootkit/main.go
++++ b/bootkit/main.go
+@@ -23,6 +23,7 @@ type tinkConfig struct {
+ 	registry string
+ 	username string
+ 	password string
++	certURL string
+ 
+ 	// Tinkerbell server configuration
+ 	baseURL    string
+@@ -38,6 +39,8 @@ type tinkConfig struct {
+ 
+ 	// Metadata ID ... plus the other IDs :shrug:
+ 	MetadataID string `json:"id"`
++
++	tinkWorkerImage string
+ }
+ 
+ func main() {
+@@ -61,7 +64,13 @@ func main() {
+ 	}
+ 
+ 	// Generate the path to the tink-worker
+-	imageName := fmt.Sprintf("%s/tink-worker:latest", cfg.registry)
++	var tinkWorkerImage string
++	if (len(cfg.tinkWorkerImage) > 0) {
++		tinkWorkerImage = cfg.tinkWorkerImage
++	} else {
++		tinkWorkerImage = "tink-worker:latest"
++	}
++	imageName := fmt.Sprintf("%s/%s", cfg.registry, tinkWorkerImage)
+ 
+ 	// Generate the configuration of the container
+ 	tinkContainer := &container.Config{
+@@ -70,6 +79,7 @@ func main() {
+ 			fmt.Sprintf("DOCKER_REGISTRY=%s", cfg.registry),
+ 			fmt.Sprintf("REGISTRY_USERNAME=%s", cfg.username),
+ 			fmt.Sprintf("REGISTRY_PASSWORD=%s", cfg.password),
++			fmt.Sprintf("REGISTRY_CERT_URL=%s", cfg.certURL),
+ 			fmt.Sprintf("TINKERBELL_GRPC_AUTHORITY=%s", cfg.grpcAuthority),
+ 			fmt.Sprintf("TINKERBELL_CERT_URL=%s", cfg.grpcCertURL),
+ 			fmt.Sprintf("WORKER_ID=%s", cfg.workerID),
+@@ -161,6 +171,9 @@ func parsecmdline(cmdlines []string) (cfg tinkConfig, err error) {
+ 			if cmdline[0] == "registry_password" {
+ 				cfg.password = cmdline[1]
+ 			}
++			if cmdline[0] == "registry_cert_url" {
++				cfg.password = cmdline[1]
++			}
+ 
+ 			// Find Tinkerbell servers settings
+ 			if cmdline[0] == "packet_base_url" {
+@@ -182,6 +195,10 @@ func parsecmdline(cmdlines []string) (cfg tinkConfig, err error) {
+ 			if cmdline[0] == "worker_id" {
+ 				cfg.workerID = cmdline[1]
+ 			}
++
++			if cmdline[0] == "tink_worker_image" {
++				cfg.tinkWorkerImage = cmdline[1]
++			}
+ 		}
+ 	}
+ 	return
+diff --git a/tink-docker/main.go b/tink-docker/main.go
+index 6e96694..d443f19 100644
+--- a/tink-docker/main.go
++++ b/tink-docker/main.go
+@@ -13,6 +13,7 @@ import (
+ 
+ type tinkConfig struct {
+ 	registry   string
++	registryCertURL string
+ 	baseURL    string
+ 	tinkerbell string
+ 
+@@ -39,7 +40,13 @@ func main() {
+ 		panic(err)
+ 	}
+ 	// Download the configuration
+-	err = DownloadFile(path+"ca.crt", cfg.baseURL+"/ca.pem")
++	var baseCertURL string
++	if (len(cfg.registryCertURL) > 0) {
++		baseCertURL = cfg.registryCertURL
++	} else {
++		baseCertURL = cfg.baseURL
```

Comment:
So baseURL here would be tinkerbell nginx IP right? 
So if we don't use a `local/private` registry and we use something like public ECR to pull `tink-worker` we technically don't need any certs right?
In this case what file would be downloaded here?

Reply:
This comment makes me want to put docker cert directory creation etc. under a conditional based on whether registry certificate path url optional parameter is present